addons/hr_recruitment/static/src/scss/hr_job.scss

```
@@ -42,10 +54,14 @@
             .o_hr_recruitment_kanban_card_large_screen {
                 display: none;    
             }
+
+            .o_kanban_record {
+                margin: 0 0 -1px;
```

Nit, just to clarify in the code what you're doing.
```suggestion
                margin: 0 0 ($border-width * -1);
```